syntax = "proto3";
option cc_enable_arenas = true;

package file1;

message Point {
	sint32 x = 1;
	sint32 y = 2;
}

message Stroke {
	repeated Point points = 1;
	sint32 width = 2;
	fixed32 color = 3;
}

message NormalLayer {
	repeated Stroke strokes = 1;
}

message Layer {
	oneof layer {
		NormalLayer normal = 1;
	}
}

message Page {
	sint32 width = 1;
	sint32 height = 2;
	repeated Layer layers = 3;
}

message File {
	repeated Page pages = 1;
}
